Bundle/BusinessEntityBundle/EventSubscriber/BusinessEntitySubscriber.php

```diff
@@ -112,40 +112,44 @@ public function updateBusinessPages($entity, BusinessEntity $businessEntity, Ent
-                );
-                // Get the BusinessPage if exists for the given entity
-                /** @var BusinessPage $businessPage */
-                $businessPage = $bepRepo->findPageByBusinessEntityAndPattern(
-                    $businessTemplate,
-                    $entity,
-                    $businessEntity
-                );
-                // If there is diff between persisted BEP and computed, persist the change
-                $scheduledForRemove = false;
-                foreach ($deletions as $deletion) {
-                    if (get_class($deletion) == get_class($businessPage)
-                        && $deletion->getId() === $businessPage->getId()
-                    ) {
-                        $scheduledForRemove = true;
+            // we generate viewRef for each BT translation
+            foreach ($businessTemplate->getTranslations() as $translation) {
+                $businessTemplate->setCurrentLocale($translation->getLocale());
+                if ($this->businessPageHelper->isEntityAllowed($businessTemplate, $entity, $entityManager)) {
+                    /** @var BusinessPageRepository $bepRepo */
+                    $bepRepo = $entityManager->getRepository('VictoireBusinessPageBundle:BusinessPage');
+                    $virtualBusinessPage = $this->businessPageBuilder->generateEntityPageFromTemplate(
+                        $businessTemplate,
+                        $entity,
+                        $entityManager
+                    );
+                    // Get the BusinessPage if exists for the given entity
+                    /** @var BusinessPage $businessPage */
+                    $businessPage = $bepRepo->findPageByBusinessEntityAndPattern(
+                        $businessTemplate,
+                        $entity,
+                        $businessEntity
+                    );
+                    // If there is diff between persisted BEP and computed, persist the change
+                    $scheduledForRemove = false;
+                    foreach ($deletions as $deletion) {
+                        if (get_class($deletion) == get_class($businessPage)
+                            && $deletion->getId() === $businessPage->getId()
+                        ) {
+                            $scheduledForRemove = true;
+                        }
                     }
-                }
 
-                if ($businessPage && !$scheduledForRemove) {
-                    $oldSlug = $businessPage->getSlug();
-                    $newSlug = $entity->getSlug();
+                    if ($businessPage && !$scheduledForRemove) {
+                        $oldSlug = $businessPage->getSlug();
+                        $newSlug = $entity->getSlug();
 
-                    $businessPage->setName($virtualBusinessPage->getName());
-                    $businessPage->setSlug($virtualBusinessPage->getSlug());
+                        $businessPage->setName($virtualBusinessPage->getName());
```

Comment:
Are you sure the name and slug of the business page will not be override by the ones of the last locale of the BusinessTemplate ? You never setup  the currentLocale of $businessPage so the second translation of the BusinessTemplate risk to erase the default locale for $businessPage